### PR TITLE
fix: add method AddPolicies and implement method AddPolicy

### PR DESCRIPTION
**Pull request by RobotHuang**
fix: https://github.com/casbin/redis-adapter/issues/9

### CODE REVIEW

**Comment by hsluoyz** [comment]
@RobotHuang plz write a normal message for commit message and PR title.

**Comment by hsluoyz** [comment]
@WesleyBatista please review.

**Comment by hsluoyz** [comment]
@RobotHuang 

Can we do the following?

1. Remove Policy(ies) API
2. Add some tests.

**Comment by RobotHuang** [comment]
> @RobotHuang
> 
> Can we do the following?
> 
> 1. Remove Policy(ies) API
> 2. Add some tests.

Yeah, I will try to do these.

**Comment by WesleyBatista** [comment]
hey there :wave:

I tried to use the code and it didn't work.

image

Currently this module points to casbin v2.1.2

I traced back the file **[parsist/batch_adapter.go](https://github.com/casbin/casbin/blob/v2.2.2/persist/batch_adapter.go)** and found out it first appears on [v2.2.2](https://github.com/casbin/casbin/blob/v2.2.2/persist/batch_adapter.go) (introduced by the PR https://github.com/casbin/casbin/pull/388).

It seems like we need to upgrade the version on **[go.mod](https://github.com/casbin/redis-adapter/blob/bf1325e218901ac64a5430fa76b73c10ccc8b8be/go.mod#L7)** to at least v2.2.2.

**Comment by hsluoyz** [comment]
@WesleyBatista nice found!

@RobotHuang this is why we need tests.

**Comment by RobotHuang** [comment]
@hsluoyz I have added Remove Policy(ies) API.

**Comment by hsluoyz** [comment]
@WesleyBatista please review.

File: go.sum [hunk 1]
```diff
@@ -2,5 +2,21 @@ github.com/Knetic/govaluate v3.0.1-0.20171022003610-9aa49832a739+incompatible h1
 github.com/Knetic/govaluate v3.0.1-0.20171022003610-9aa49832a739+incompatible/go.mod h1:r7JcOSlj0wfOMncg0iLm8Leh48TZaKVeNIfJntJ2wa0=
 github.com/casbin/casbin/v2 v2.1.2 h1:bTwon/ECRx9dwBy2ewRVr5OiqjeXSGiTUY74sDPQi/g=
```
Comment (hsluoyz) [inline go.sum:3]: @yyellowsun can you run `go mod tidy` to clean the go.sum file?

**Comment by WesleyBatista** [comment]
It seems to be good! 

I'm not so experience with golang so my approach to test this was to verify if things were running as expected.

I added my piece of code from the issue #9 and ran the test:

image


And as expected it failed, with the following error message:
```
--- FAIL: TestAdapter (111.87s)
    adapter_test.go:115: Policy:  [[alice data1 read] [bob data2 write] [data2_admin data2 read] [data2_admin data2 write] [admin domain1 *] [anonymous * /]] , supposed to be  [[alice data1 read] [bob data2 write] [data2_admin data2 read] [data2_admin data2 write]]
FAIL

```

After the failure, the result on redis was looking like this:

image
_using [redis-commander tool](https://github.com/joeferner/redis-commander) to visualize redis data_

And we can see that:
 - **paul** gets added and disappears after `RemovePolicy` call :heavy_check_mark: 
 - **curry** and **kobe** were removed by the newly implemented `RemovePolicies` :heavy_check_mark:
 -  **admin** and  **anonymous** exist on redis after `AddNamedPolicies` call :heavy_check_mark: 




Thanks for the implementation @RobotHuang! Also very nice addition of yours to use more appropriate redis data types for this.

Happy to see the coverage even increased +0.9%. Just a little bit but still :muscle:

**Review by WesleyBatista — APPROVED** [review]

**Pull request merged**